File: Number_Theory_Functions/Get_Factors/get_factors.cpp
# include <iostream>
# include <vector>
# include <cmath>
using namespace std;

vector<int> get_factors(int n){
    vector<int> ans;
    int root = sqrt(n);
    for(int i=1; i<=root;i++){
        if (n%i ==0){
            ans.push_back(i); //pushing the factors into a vector
            if ( i != (n/i)){
                ans.push_back(n/i); // pushing the corresponding factors which are greater than root(n)
            }

        }
    }
    return ans;
}

int main(){
    vector<int> a = get_factors(20); //sample input 1
    vector<int> a = get_factors(4); // sample input 2
    return 0;
}

File: Number_Theory_Functions/Get_Factors/get_factors.java
import java.util.ArrayList;

public class get_factors{
    public static ArrayList<Integer> get_factor(int n){
        ArrayList<Integer> ans = new ArrayList<>();
        double root = Math.floor(Math.sqrt(n));
        for (int i=1; i<=root;i++){
            if ( n%i ==0){
                ans.add(i); //pushing the factors <= (root) into an ArrayList
                if ( i != (n/i)){
                    ans.add(n/i);  // pushing the corresponding factors which are greater than root(n)
                }
            }
        }
        return ans;
    }

    public static void main(String args[]){
        System.out.println(get_factor(4)); // sample input 1
        System.out.println(get_factor(20)); // sample input 2
    }
}
